### ISSUE TITLE
supports-emoji - supports-color for cli emoji

### ISSUE THREAD

**Issue by dylang** [sindresorhus/module-requests#64]
### Today

On a Mac, it's easy enough to use emoji such as ❤️, just by outputting that character to the console. (That's a unicode character, not an image.)

### Problem 

* Most of the Linux distros don't support them well by default.
* Most of the ssh gateways don't support them at all; e.g: continuous integration/deployment.
* I'm not sure about Windows, but I'm guessing support isn't there or isn't great.

### Solution

`supports-emoji` library, similar to [supports-color](https://github.com/chalk/supports-color).

```js
const supportsEmoji = require('supports-emoji');
console.log(`I ${supportsEmoji ? '❤️' : '<3'} your work!`);
```

_Note: it looks like Github code blocks don't support emoji. That block is a ❤️ too._

**Comment by dylang** [comment]
With zero testing (I don't have all the necessary environments available to me), I'm wondering if something like this might work:

```js
const supportsColor = require('supports-color');

// this is just a guess that terms without good color support probably also can't do emoji
const supportsEmoji = supportsColor.level > 2;
```

**Comment by ewnd9** [comment]
```js
const supportsEmoji = supportsColor.level > 2;
```

shows me `false` in `gnome-terminal` in Ubuntu 14.04. 

http://probablement.net/txt/emojilinux - related article about state of emoji support on Linux systems

**Comment by dylang** [comment]
Thanks @ewnd9. Is `false` correct or wrong for your terminal?

**Comment by forivall** [comment]
I wouldn't suggest using supports-color, if you look at the source, you'll just see that, for 16m color, the only way to say that you support it is through command line arguments and nothing in termcap.

For supports emoji, I would suggest the following levels: 0: nope. 1: `require('has-unicode')()`. 2: monochrome only 3: full color. I would be 2, since I have emoji fonts installed, and most OSX would be 3.

For detection on linux, this would probably require checking if it's a local terminal, and then checking fontconfig to see if an emoji font is installed (symbolo, noto, etc). Or, once distros start supporting emoji explicitly, a simple `uname` or `lsb_release` would give the answer. A quick google search didn't show any linux terminal emulators supporting full-color emoji. I wonder if there'll be anything added to termcap to indicate emoji support...

**Comment by ewnd9** [comment]
@dylang yes, it is correct, I have only monochrome emoji with [emojione-color-font](https://github.com/eosrei/emojione-color-font#install-on-linux) installed.

Firefox has full-color emoji support, so it could be used to test with browser terminal emulators 

![image](https://cloud.githubusercontent.com/assets/1713789/14059533/0885abe4-f37f-11e5-9f20-78bdc17f9d4f.png)

```
const supportsEmoji = supportsColor.level > 2;
```

Also shows `false` there, so for firefox case, it is not correct.

**Comment by dylang** [comment]
Oops, had a typo, it should have been `supportsColor.level >= 2`, but it still sounds like this is not a useful metric.

**Comment by Qix-** [comment]
Some emulators define the `TERM_PROGRAM` environment variable. Just a thought.

**Comment by Richienb** [comment]
At this point in time, there's not a definite way of checking for emoji support since there are major variables in how they are supported (fonts, built-in, etc.).

However, if such is possible, the tool should return an integer from 0-3 depending on its support:

0. Unicode not supported
1. Unicode supported
2. Monochrome emoji supported
3. Colour emoji supported

**Comment by Richienb** [comment]
Development started: https://github.com/Richienb/supports-emoji
